### PR TITLE
fix trusted_proxies for caddy in case of a reverse proxy situation

### LINKED ISSUE #828: trusted proxies does not work as intended
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

### Steps to reproduce
1. Install nextcloud aio with custom apache port
2. Nextcloud brute force thinks every access comes from ::1 and the log thinks every wrong login comes from ::1 / `docker logs nextcloud-aio-nextcloud` shows 172.25.0.9
3. Set ::1 as trusted proxy in the config.php
4. With apache and caddy fd00:dead:beef:c0::1 is now shown / with the Nginx Proxy Manager 172.17.0.1, both are the gateways of the docker bridge network (X-Forwarded-For header is set in all the configurations)
5. Setting fd00:dead:beef:c0::1/172.17.0.1/172.25.0.9 as trusted proxy changes nothing.

### Expected behavior 
The real IP is shown (which is sent in the X-Forwarded-For header)
### Actual behavior 
::1 is shown as IP to access / `docker logs nextcloud-aio-nextcloud` shows 172.25.0.9
With ::1 set as trusted proxy, it shows fd00:dead:beef:c0::1  /  `docker logs nextcloud-aio-nextcloud` shows 172.25.0.9
Setting fd00:dead:beef:c0::1/172.17.0.1/172.25.0.9 as trusted proxy changes nothing
This makes the brute force protection useless and makes it block everything, instead of a certain IP
If my cloud gets blocked, I need to run `[sudo] docker exec -it nextcloud-aio-nextcloud php occ security:bruteforce:reset ::1` to make it possible for me to login, (Whitelisting ::1 in brute force settings app also works)

### Host OS 
Debian GNU/Linux 11 (bullseye)

#### Nextcloud AIO version 
v1.4.0  and v1.4.1
#### Current channel 
Beta
#### Other valuable info 
Default Nextcloud config.php (Apache/Caddy and NPM):
grafik
NPM with ::1 as set trusted proxy:
grafik
Apache2/Caddy with ::1 set as trusted proxy:
grafik

fd00:dead:beef:c0::1 and 172.17.0.1 are the gateways of the docker bridge network and 172.25.0.9 the IP of the nextcloud-aio-apache container
The X-Forwarded-For header is set by all three proxies to my real IP (`185.***.**.**`)

Edit:
Also, this block of trusted proxies does not work (And yes, some entries in this block are nonsense)
```
  array (
    0 => '127.0.0.1',
    1 => '172.25.0.9:*',
    2 => '172.17.0.1:*',
    3 => '0.0.0.0/0:*',
    4 => '0.0.0.0/0',
    5 => '0.0.0.0',
    6 => '::1/0:*',
    7 => '::1/0',
    8 => '::1',
  ),
```

### PR DESCRIPTION
**Pull request by szaimen**
Close #828

Signed-off-by: szaimen <szaimen@e.mail.de>

### CODE REVIEW

File: Containers/apache/Caddyfile [hunk 1]
```diff
@@ -4,6 +4,9 @@
     storage file_system {
         root /mnt/data/caddy
     }
+
+    # See https://github.com/nextcloud/all-in-one/issues/828
+    # trusted_proxies placeholder
```
Comment (2020Sanoj) [inline Containers/apache/Caddyfile:9]: grafik
I think this does not work here

Comment (szaimen) [reply Containers/apache/Caddyfile:9]: I see, that is unfortune but okay if it works

Comment (szaimen) [reply Containers/apache/Caddyfile:9]: hm.... this is getting worse then I thought since we need brackets now which are hard to parse and modify via sed...

Comment (szaimen) [reply Containers/apache/Caddyfile]: okay, seems like empty brackets are still accepted so not as problematic as I thought.

**Review by 2020Sanoj — APPROVED** [review]
Should work (If the proxy connects over localhost / an IP from the private IP range)

I've tested manipulating the X-forwarded-for header using a local reverse proxy and a local DNS rewrite (`dig mydomain` showed the DNS rewrite and my browser was inside the logs of the local proxy)
And the manipulating didn't work, Nextcloud showed me the external IP, which my local proxy used to connect to my Nextcloud. So it should be not possible to trick Caddy with a manipulated X-forwarded-for header. 
But this also means that if nextcloud-aio is used behind a reverse proxy, which is not connecting to nextcloud-aio using a local / private range IP, the X-forwarded-for header wouldn't work.

**Comment by szaimen** [comment]
Thanks for the review and approval! :)
